### ISSUE TITLE
Add typescript types

### ISSUE THREAD

**Issue by gbersac** [mrbar42/node-graceful#1]
Hi, your package is exactly what I am looking for, but I am using typescript and there seems not to be types associated with your package. I tried `npm install @types/node-graceful` but there is no such package. Please you or someone else create one !

**Comment by mrbar42** [comment]
Thanks for opening an issue!
great idea, I'll try to find time and do that or someone might beat me to it :smile:

**Comment by mrbar42** [comment]
@gbersac checkout version `v0.2.4`

**Comment by gbersac** [comment]
@mrbar42 Great, that was quick ! For typescript users to know that there is types available for this package, you should release the types so that they are accessible using `npm i @type/node-graceful --save-dev`. That's what they'll type when they will be looking for the types of this module ^^

**Comment by mrbar42** [comment]
As far as i understand having a d.ts in the project itself achieves the
same goal isn't it?

**Comment by gbersac** [comment]
@mrbar42 I don't really know. All I know is that, if I type `npm i @type/node-graceful `, it doesn't find the package. But maybe it is because you didn't published your update to npm.

**Comment by yss14** [comment]
Quote from [DefinitelyTyped](https://github.com/DefinitelyTyped/DefinitelyTyped): 

> If you are the library author, or can make a pull request to the library, bundle types instead of publishing to DefinitelyTyped.

But for me the typings unfortunately don't work: 
```
Could not find a declaration file for module 'node-graceful'. '<path>/node_modules/node-graceful/graceful.js' implicitly has an 'any' type.
  Try `npm install @types/node-graceful` if it exists or add a new declaration (.d.ts) file containing `declare module 'node-graceful';`

```

**Comment by mrbar42** [comment]
@yss14 Thanks for sharing the error, it turns out typescript expects the `d.ts` file to be named `index.d.ts` which was fixed in #2 

checkout the latest version (`v0.3.0`) and share any further problems

**Comment by awwong1** [comment]
```
[ts] Generic type 'Promise<T>' requires 1 type argument(s).
interface Promise<T>
var Promise: PromiseConstructor
Represents the completion of an asynchronous operation
```

Need to add `<any>` next to Promise.

**Comment by awwong1** [comment]
Thanks for merging this in! I added in some code in that PR as a comment.
I think you should be able to close this issue once you create a new release.

**Comment by mrbar42** [comment]
should be fixed in [v0.3.1](https://github.com/mrbar42/node-graceful/releases/tag/v0.3.1) 🎉 

please reopen if needed
